packages/validate/test/ValidateCoreMixin.test.js

```diff
@@ -0,0 +1,1178 @@
+      const el = await fixture(html`
+        <${tag}
+          .validators=${[new Required()]}
+          .modelValue=${''}
+        >${lightDom}</${tag}>
+      `);
+      const validator = el.validators.filter(v => v instanceof Required);
+      const executeSpy = sinon.spy(validator, 'execute');
+      const privateIsEmptySpy = sinon.spy(el, '__isEmpty');
+      expect(executeSpy.callCount).to.equal(0);
+      expect(privateIsEmptySpy.callCount).to.equal(1);
+    });
+
+    it('calls "._isEmpty" when provided, for different modelValues', async () => {
+      const customRequiredTagString = defineCE(
+        class extends ValidateCoreMixin(LitElement) {
+          _isEmpty(modelValue) {
+            return modelValue.model === '';
+          }
+        },
+      );
+      const customRequiredTag = unsafeStatic(customRequiredTagString);
+
+      const el = await fixture(html`
+        <${customRequiredTag}
+          .validators=${[new Required()]}
+          .modelValue=${{ model: 'foo' }}
+        >${lightDom}</${customRequiredTag}>
+      `);
+
+      const providedIsEmptySpy = sinon.spy(el, '_isEmpty');
+      el.modelValue = { model: '' };
+      expect(providedIsEmptySpy.callCount).to.equal(1);
+      expect(el.errorStates.required).to.be.true;
+    });
+
+    it('prevents other Validators from being called when input is empty', async () => {
+      const alwaysInvalid = new AlwaysInvalid();
+      const alwaysInvalidSpy = sinon.spy(alwaysInvalid, 'execute');
+      const el = await fixture(html`
+        <${tag}
+          .validators=${[new Required(), alwaysInvalid]}
+          .modelValue=${''}
+        >${lightDom}</${tag}>
+      `);
+      expect(alwaysInvalidSpy.callCount).to.equal(0); // __isRequired returned false (invalid)
+      el.modelValue = 'foo';
+      expect(alwaysInvalidSpy.callCount).to.equal(1); // __isRequired returned true (valid)
+    });
+
+    it('adds [aria-required="true"] to "._inputNode"', async () => {
+      const el = await fixture(html`
+        <${tag}
+          .validators=${[new Required()]}
+          .modelValue=${''}
+        >${lightDom}</${tag}>
+      `);
+
+      expect(el.getAttribute('aria-required')).to.equal('true');
+      el.modelValue = 'foo';
+      expect(el.getAttribute('aria-required')).to.equal('false');
+    });
+  });
+
+  describe('Preconfigured Validators', () => {
+    it('Can store preconfigured Validators for custom inputs', async () => {
+      const defaultRequiredTagString = defineCE(
+        class extends ValidateCoreMixin(LitElement) {
+          constructor() {
+            super();
+            this._preconfiguredValidators = [new Required()];
+          }
+        },
+      );
+      const defaultRequiredTag = unsafeStatic(defaultRequiredTagString);
+
+      const el = await fixture(html`
+      <${defaultRequiredTag}
+        .validators=${[new MinLength(3)]}
+      >${defaultRequiredTag}</${tag}>`);
+
```

Comment:
make `_preconfiguredValidators` public => so people could override messages?
```js
 const el = await fixture(html`
      <${tag}
        .validators=${[new MinLength(3)]}
        .preconfiguredValidators=${[new Required(null, { message: () => return 'My Error message' })]}
      ></${tag}>`);
```
or individually `el.preconfiguredValidators[0].options.message` ???? hmm these are classes? so maybe not?

add 
```js
get allValidators() {
  return [...this.preconfiguredValidators, ...this.validators];
}
```

so `.validators` stays pure to what the users sets